src/equations/numerical_fluxes.jl

```diff
@@ -305,12 +307,32 @@ end
 
 @inline function (numflux::FluxHydrostaticReconstruction)(u_ll, u_rr,
                                                           orientation_or_normal_direction,
-                                                          equations::AbstractEquations)
+                                                          equations::AbstractEquations{1})
   @unpack numerical_flux, hydrostatic_reconstruction = numflux
 
   # Create the reconstructed left/right solution states in conservative form
   u_ll_star, u_rr_star = hydrostatic_reconstruction(u_ll, u_rr, equations)
 
+  # Set the water height to be at least the value stored in the variable threshold after
+  # the hydrostatic reconstruction is applied and before the numerical flux is calculated
+  # to avoid numerical problem with arbitrary small values. Interfaces with a water height
+  # lower or equal to the threshold can be declared as dry.
+  # The default value is set to 1e-15 and can be changed within the constructor call in an elixir.
+  threshold = equations.threshold_wet
+
+  # Apply threshold to cut off the height and the velocity at dry interfaces
+  if threshold > 0
+    h_ll, _, b_ll = u_ll_star
+    h_rr, _, b_rr = u_rr_star
+
+    if h_ll <= threshold
+      u_ll_star = SVector(threshold, zero(eltype(u_ll)), b_ll)
+    end
+
+    if h_rr <= threshold
+      u_rr_star = SVector(threshold, zero(eltype(u_rr)), b_rr)
+    end
+  end
```

Comment:
I would move this dimension specific cutoff into the `hydrostatic_reconstruction` routine of Chen & Noelle. That way this wrapper `FluxHydrostaticReconstruction` is generic and can dispatch only on `AbstractEquations`.

This will also fix some of the failing tests.